src/connector/src/source/iceberg/parquet_file_handler.rs

```diff
@@ -110,14 +110,30 @@ pub fn new_s3_operator(
     s3_secret_key: String,
     bucket: String,
 ) -> ConnectorResult<Operator> {
-    // Create s3 builder.
-    let mut builder = S3::default().bucket(&bucket).region(&s3_region);
-    builder = builder.secret_access_key(&s3_access_key);
-    builder = builder.secret_access_key(&s3_secret_key);
-    builder = builder.endpoint(&format!(
-        "https://{}.s3.{}.amazonaws.com",
-        bucket, s3_region
-    ));
+    let is_minio = s3_region.starts_with("http");
```

Comment:
Here is the executor. I think it's better to put this logic at src/frontend/src/optimizer/rule/table_function_to_file_scan_rule.rs

Reply:
It won't work to put it in the `table_function_to_file_scan_rule`. I understand that you want this logic to be completed in the frontend, but the FileScanExecutor is constructed [directly from the proto](https://github.com/risingwavelabs/risingwave/blob/2601c237a7dd677c1314482f91720f4ea78841b5/src/batch/executors/src/executor/s3_file_scan.rs#L120-L131), and modifying it would require changes to the proto. Therefore, I suggest we maintain the current implementation.

Reply:
I think changing the proto is acceptable, because it is a batch proto, so no breaking change.